example/lib/named_routes.g.dart

```diff
@@ -81,3 +81,37 @@ extension $PersonRouteExtension on PersonRoute {
     buildContext.go(location(queryParams: queryParams), extra: extra);
   }
 }
+
+GoRoute get loginGoRoute => RouteData.$route(
+      path: '/login',
+      factory: $LoginRouteExtension._fromState,
+    );
+
+extension $LoginRouteExtension on LoginRoute {
+  static LoginRoute _fromState(GoRouterState state) => LoginRoute(
+        from: state.queryParams['from'],
+      );
+
+  String location({Map<String, String>? queryParams}) => RouteData.$location(
+        '/login',
+        //TODO: skip the `?` in the generated path if queryParams is empty???
+        queryParams: queryParams == null && from == null
```

Comment:
I noticed that if you have `queryParams` empty, (but not null) then the path has a `?` in it, which messes up the comparison logic. Should look at this @csells